Update Swarm integration guide and make it an official part of the docs

aanand opened the pull request:
Fixes #2656

A step towards removing the warning labels from Compose/Swarm integration.

dnephin commented:
I think you need to use absolute paths for cross-product links, to fix the `documentation` build.

dnephin commented on docs/swarm.md:
```diff
@@ -0,0 +1,81 @@
+documentation](/swarm/scheduler/filter/).
+
+
+## Limitations
```
We might need to mention that `volumes_from` and `network_mode: service:x` still co-schedule containers, which breaks any applications that have dependencies like: `A -> C, B -> C`.

dnephin commented on docs/swarm.md:
```diff
@@ -0,0 +1,149 @@
+
+-   `volumes_from`
+
+-   `links` (in the version 1 file format only).
```
as far as I've heard this is still true for v2 as well.

aanand commented:
Updated:

- Removed the inaccurate "(in v1 only)" comment about `links`
- Added section on the "unable to find a node that satisfies container=abc123..." problem and workarounds

SvenDowideit commented:
@aanand this needs a rebase, the tutorials project went private, and the fix is in master :)

aanand commented:
@SvenDowideit done!

GordonTheTurtle commented:
Job: Compose-PRs [FAILED](https://jenkins.dockerproject.org/job/Compose-PRs/2370/console):

~~~console

---


compose/state                      0      0      0      0   100%   

compose/utils                     53      1     20      1    97%   17

compose/volume                    67      3     18      3    93%   79-80, 99

--------------------------------------------------------------------------

TOTAL                           2903    380   1036    184    86%   

Coverage HTML written to dir coverage-html

=========================== short test summary info ============================

XFAIL tests/unit/config/config_test.py::VolumePathTest::test_split_path_mapping_with_windows_path

  does not have a drive

XFAIL tests/unit/config/types_test.py::TestVolumeSpec::()::test_parse_volume_windows_absolute_path

  does not have a drive

SKIP [1] tests/unit/config/config_test.py:1615: windows paths



============== 623 passed, 1 skipped, 2 xfailed in 475.50 seconds ==============

py34 develop-inst-nodeps: /code

py34 installed: cached-property==1.2.0,coverage==3.7.1,docker-compose==1.7.0.dev0,docker-py==1.8.0.dev0,dockerpty==0.4.1,docopt==0.6.1,enum34==1.0.4,jsonschema==
---

~~~

dnephin commented on docs/production.md:
```diff
@@ -69,14 +69,13 @@ commands will work with no further configuration.
-in beta, but if you'd like to explore and experiment, check out the <a
-href="https://github.com/docker/compose/blob/master/SWARM.md">integration
-guide</a>.
+in beta, but if you'd like to explore and experiment, check out the [integration
```
swarm isn't beta anymore!

dnephin commented on docs/swarm.md:
```diff
@@ -0,0 +1,179 @@
+      baz:
+        image: baz
+        environment:
+          - "constraint:node==node-1"
```
It seems strange that this manual scheduling example is up here instead of in the manual scheduling section.

aanand replied:
I think it makes sense to follow on from the previous (non-working) example to the same app with the workaround applied. It would be good to cover the "specific node" use case in the manual scheduling section though, so I've added an example there.

dnephin commented:
LGTM, just minor comments

GordonTheTurtle commented:
Job: Compose-PRs [FAILED](https://jenkins.dockerproject.org/job/Compose-PRs/2374/console):

~~~console

---
......Passed

Trim Trailing Whitespace.................................................Failed

hookid: trailing-whitespace



Fixing docs/compose-file.md



Reorder python imports...................................................Passed

ERROR: InvocationError: '/code/.tox/pre-commit/bin/pre-commit run --all-files'

___________________________________ summary ____________________________________

ERROR:   pre-commit: commands failed

Build step 'Execute shell' marked build as failure
[PostBuildScript] - Execution post build scripts.
[Compose-PRs] $ /bin/sh -xe /tmp/hudson3260729505099619788.sh
+ set +x


---

~~~

dnephin commented:
LGTM

GordonTheTurtle commented:
Job: Compose-PRs [FAILED](https://jenkins.dockerproject.org/job/Compose-PRs/2375/console):

~~~console

---
......Passed

Trim Trailing Whitespace.................................................Failed

hookid: trailing-whitespace



Fixing docs/compose-file.md



Reorder python imports...................................................Passed

ERROR: InvocationError: '/code/.tox/pre-commit/bin/pre-commit run --all-files'

___________________________________ summary ____________________________________

ERROR:   pre-commit: commands failed

Build step 'Execute shell' marked build as failure
[PostBuildScript] - Execution post build scripts.
[Compose-PRs] $ /bin/sh -xe /tmp/hudson5111426128910561536.sh
+ set +x


---

~~~
